src/functional/ffront/foast_passes/utils.py

```diff
@@ -0,0 +1,47 @@
+from functional.ffront import field_operator_ast as foast
+
+
+def compute_assign_indices(
+    targets: list[foast.FieldSymbol | foast.TupleSymbol | foast.ScalarSymbol | foast.Starred],
+    num_elts: int,
+) -> list[tuple[int, int] | int]:
+    """
+    Compute a list of relative indices, mapping each target to its respective value(s).
+
+    This function is used when mapping a tuple of targets to a tuple of values, and also handles the
+    case in which tuple unpacking is done using a Starred operator.
+
+    Examples
+    --------
+    The below are examples of different types of unpacking and the
```

Comment:
You can keep this as is, but the line break above is wrapped at 100 characters, but much less here.